File: src/201-300/217. Contains Duplicate/Solution.js
// More Efficient Solution
/**
 * @param { number[] } nums
 * @return { boolean }
 */
const containsDuplicate = function (nums) {
	const nums_set = new Set()

	for (const n of nums) {
		if (nums_set.has(n)) {
			return true
		}

		nums_set.add(n)
	}

	return false
}


// Less Efficient Solution
/**
 * @param { number[] } nums
 * @return { boolean }
 */
const containsDuplicate2 = function (nums) {
	return (new Set(nums)).size !== nums.length
}

File: src/201-300/217. Contains Duplicate/Solution.py
from typing import List


# More Efficient Solution
class Solution:
    @staticmethod
    def containsDuplicate(nums: List[int]) -> bool:
        num_set = set()

        for n in nums:
            if n in num_set:
                return True

            num_set.add(n)

        return False


# Less Efficient Solution
class Solution2:
    @staticmethod
    def containsDuplicate(nums: List[int]) -> bool:
        return len(set(nums)) != len(nums)
